src/rust/engine/process_execution/src/remote.rs

```diff
@@ -86,20 +93,19 @@ impl CommandRunner {
                 .map(|result| (Arc::new(execute_request), result))
           })
 
-          // TODO: Use some better looping-frequency strategy than a tight-loop:
-          // https://github.com/pantsbuild/pants/issues/5503
-
           // TODO: Add a timeout of some kind.
           // https://github.com/pantsbuild/pants/issues/5504
 
           .and_then(move |(execute_request, operation)| {
             future::loop_fn(operation, move |operation| {
               match extract_execute_response(operation) {
                 Ok(value) => {
+                  // backoff_per_operation.remove(operation.clone().get_name());
```

Comment:
the goal is to remove the backoff period tracking once the operation is `Ok`ed or `Err`ed, but I'm getting
```
error[E0382]: use of moved value: `operation`
   --> process_execution/src/remote.rs:103:49
    |
101 |               match extract_execute_response(operation) {
    |                                              --------- value moved here
102 |                 Ok(value) => {
103 |                    backoff_per_operation.remove(operation.clone().get_name());
    |                                                 ^^^^^^^^^ value used here after move
    |
    = note: move occurs because `operation` has type `bazel_protos::operations::Operation`, which does not implement the `Copy` trait
```
wonder if there's a good way to access `operation.name` in the context.

Reply:
or maybe not so important to reclaim the memory if `run_command_remote`'s context is small enough?

Reply:
If you wanted to do this, what I'd suggest is adding a `let operation_name = operation.get_name().clone();` on the line before the match, and using `operation_name` here; the problem is that `extract_execute_response` takes ownership of the Operation's memory. An alternative would be to make `extract_execute_response` return a `Ok((name, value))` instead of just an `Ok(value)` which you could extract from the match.